src/components/ShoppingCart/ShoppingCart.js

```diff
@@ -0,0 +1,88 @@
+import { useContext } from "react";
+import { Link } from "react-router-dom";
+import CartContext from "../../state/CartContext";
+import CartItem from "../CartItem/CartItem";
+import { useCartProducts } from "../../utils/hooks/useCartProducts";
+
+function ShoppingCart() {
+  const { state, dispatch } = useContext(CartContext);
+  const { data: cartProducts, isLoading: cartProductsLoading } =
+    useCartProducts(state.cart.map((item) => item.id));
+
+  function onChangeQuantity(id, quantity) {
+    dispatch({
+      type: "UPDATE_CART",
+      payload: {
+        id,
+        quantity,
+      },
+    });
+  }
+
+  function onRemoveItem(id) {
+    dispatch({ type: "REMOVE_FROM_CART", payload: id });
+  }
+
+  function getStock(product) {
+    return cartProducts.results?.find((item) => item.id === product.id).data
+      .stock;
+  }
+
+  function calculateTotal() {
```

Comment:
Do you think is possible or more useful to have this on your reducer?

Reply:
Yup, got it!